shopfloor_reception/services/reception.py

```diff
@@ -256,13 +256,8 @@ def _scan_line__find_or_create_line(self, picking, move, qty_done=1):
                 )
             )
         )
-        if line:
-            # The line quantity to do needs to correspond to
-            # the remaining quantity to do of its move.
-            line.product_uom_qty = move.product_uom_qty - move.quantity_done
-        else:
-            qty_todo_remaining = max(0, move.product_uom_qty - move.quantity_done)
-            values = move._prepare_move_line_vals(quantity=qty_todo_remaining)
+        if not line:
+            values = move._prepare_move_line_vals()
```

Comment:
Shouldn't there not a kind of lock?
If users work in parrallel, they will get serialize errors.
Because in the method _scan_line__assign_user the picking.user_id will be set and also of course the shopfloor_user_id of the stock.move.line. 
Also i think we should remove the self._assign_user_to_picking from _scan_line__assign_user

If a line was found, shouldn't we change the product_uom_qty to 0. Otherwise _auto_post_line can maybe make some trouble if we work in parallel.

Reply:
I think removing `self._assign_user_to_picking `   from _scan_line__assign_user is a good idea because the user assigned to picking is not used in this scenario.
@jbaudoux  ?

About reseting `product_uom_qty` to zero it is probably already done enough in this scenario (Before line is posted, all are reset)

Reply:
I mean if a user starts and there is already an unassigned move.line where product_uom_qty is already set. I think it would be safe to set it to 0, then it would be the same as creating a new move.line

Reply:
If two users start a the same time with the same product this will cause an serialize access error, because the will both get maybe the same existing move.line if there is one.
And if we are not remove the assign_user_to_picking this will cause also an serialize access error.

Reply:
This function has been improved by the commit [shopfloor_reception: improve move line assignation to user](https://github.com/OCA/wms/pull/726/commits/d56d148c0feddef38fe6045852f1e29acb98b2f0)
First to lock a row being assigned to the user.
Second the first implementation missed some sorting and the first line being choosen could have been one without shopfloor_user assigned although a line with the current user assigned does exist.